### ISSUE TITLE
[Request] Order of response attributes

### ISSUE THREAD

**Issue by MorningSleeper** [iexg/IEX-API#1311]
In the API docs I noticed the order of the response attributes where they are described, section labeled "Response Attributes", does not always match the order of the response attributes in the API's output or the JSON Response section in API docs that shows an example of the API output such as in [quote ](https://iexcloud.io/docs/api/#quote )endpoint.

Is the order supposed to match or is that by design and if so what is the method by which they are ordered in each section? Regardless of the way they are ordered, it would be helpful if in the future you could make the orders match so that when one goes from one section to the other one will know where in the list to expect to find the same response attribute.

Edited, to clarify question.

**Comment by McQgit** [comment]
@MorningSleeper IEX (evan?) has said in the past that the order can vary and is considered normal.  FWIW, my json parser isn't impacted by the sequence - I access the parsed array by response item name.

**Comment by MorningSleeper** [comment]
@McQgit thanks for reply, as you mention, the sequence of the response attributes returned by the endpoint does not matter since most would access them by the response attribute names.

I think you understood what I was asking but just to clarify for anyone else, I was referring to the two sections in the API docs for each endpoint - the one titled "JSON Response" and the one below it titled "Response Attributes". While I was curious if there was any method to the ordering of the response attributes in each section, I mostly wanted to pass along the idea that, regardless of their order, it would be helpful to have the order of the response attributes in each section match so that one can go back and forth between the different response attribute names/descriptions in the lower section to the response attribute example data in the JSON Response section above it and find them in the same sequence.

**Comment by McQgit** [comment]
@MorningSleeper Oh, I see now.  You want the docs changed - not the api data.  Nevermind then.

FWIW, I got led astray by your 'data' category in the title and saying 'quote endpoint' so thought it was api data related.

**Comment by MorningSleeper** [comment]
Essentially, I suspect however that neither section of API docs is dynamcially generated and thus not something that could be easily updated at least at this time. I have changed the subject to make it a feature request/suggestion for a future version of the API.

So the specific suggestion is, if possible, consider making the order of each endpoint's API response attributes that are returned by the API and reflected in the example data in the JSON Response section of API docs to match the order of the response attributes in the Response Attributes section of API docs. The actual order does not matter as much as the consistency of the order between all three places although it would also be nice to settle on an order method that is consistent between endpoints like alphabetical by response attribute name or something,

**Comment by timpaine-iex** [comment]
JSONs are fundamentally unordered though.

**Comment by MorningSleeper** [comment]
@timpaine-iex I understand, I made things confusing by mentioning the JSON returned by the endpoint which as you mention is unordered.

My suggestion should have been limited to the two sections in the API docs for each endpoint - the one titled "JSON Response" that shows an example and the one below it titled "Response Attributes". 

It would be helpful to have the order, whatever it may be, of the response attributes in each section match so that one can go back and forth between the different response attribute names & descriptions in the lower section to the response attribute example data in the JSON Response section above it and find them in the same sequence.

At the time of my post this was not the case in the quote example I originally provided and some others but I see the quote one has since been fixed.
